qa/tasks/cephadm.py

```diff
@@ -364,11 +364,12 @@ def ceph_bootstrap(ctx, config, registry):
             '--output-keyring',
             '/etc/ceph/{}.client.admin.keyring'.format(cluster_name),
             '--output-pub-ssh-key', '{}/{}.pub'.format(testdir, cluster_name),
+            '--mon-id', first_mon,
+            '--mgr-id', first_mgr,
```

Comment:
```
2020-07-11T05:43:03.562 INFO:tasks.cephadm:Teardown complete
2020-07-11T05:43:03.562 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 368, in ceph_bootstrap
    '--mgr-id', first_mgr,
UnboundLocalError: local variable 'first_mgr' referenced before assignment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 93, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 1217, in task
    lambda: cephfs_setup(ctx=ctx, config=config),
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 55, in nested
    raise exc[1]
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 1095, in initialize_config
    yield
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 31, in nested
    vars.append(enter())
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 471, in ceph_bootstrap
    ctx.daemons.get_daemon(type_, id_, cluster).stop()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/daemon/cephadmunit.py", line 149, in stop
    self.remote.sh(self.stop_cmd)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 248, in sh
    proc = self.run(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 204, in run
    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
```

https://pulpito.ceph.com/kchai-2020-07-11_05:07:55-rados-wip-kefu-testing-2020-07-11-1109-distro-basic-smithi/

Reply:
hm. `--mon-id` and `--mgr-id` only make sense in roleless mode. We#re leaving the roleless mode when we require the `-mon-id` param. 

I'd say either we have to use the non-roleless mode or we have to fix the tests.

Reply:
ok this change doesn't really make sense. can you revert this mon-ip thingy, please?